src/lib/useToken.js

```diff
@@ -0,0 +1,12 @@
+import { useState, useEffect } from 'react';
+import { USER_TOKEN } from '../../src/constants';
+
+export function useToken() {
+  const [userToken, setToken] = useState(() =>
+    localStorage.getItem(USER_TOKEN),
+  );
+
+  useEffect(() => setToken(userToken), [userToken]);
```

Comment:
We're pretty uncertain about this . . .

Reply:
Oh that's funny. I ended up doing something similar in a separate experimental branch. Same general idea, but mine might be more awkward.